sirepo/package_data/static/js/sirepo-command.js

```diff
@@ -286,14 +293,28 @@ SIREPO.app.directive('commandTable', function(appState, commandService, latticeS
                 if (! data) {
                     return;
                 }
-                var i = commandIndex(data);
-                data = $scope.commands.splice(i, 1)[0];
-                if (i < index) {
-                    index--;
+                if (data.sim_type !== SIREPO.APP_SCHEMA.simulationType) {
+                    return;
+                }
+                if (data.sim_id !== appState.models.simulation.simulationId) {
+                    // item dropped from another simulation window
+                    data.command._id = latticeService.nextId();
+                    data._id = data.command._id;
+                    data.sim_id = appState.models.simulation.simulationId;
+                    appState.models.commands.splice(index, 0, data.command);
+                }
+                else {
+                    const prevIndex = commandIndex(data);
+                    if (prevIndex < 0 || prevIndex === index) {
+                        return;
+                    }
+                    $scope.commands.splice(prevIndex, 1);
                 }
                 $scope.commands.splice(index, 0, data);
+                $scope.selectItem(data);
                 saveCommands();
             };
+
             // expects a negative number to move up, positive to move down
             $scope.moveItem = function(direction, command) {
```

Comment:
As long as this directive is being changed, might as well use arrow syntax for existing functions for consistency:
 ```
$scope.moveItem = (direction, command) => {...}
```
etc.

Reply:
Yes, there are many cases of the old function arg format, but I didn't change it for moveItem() in this case because I didn't modify that code. We could have a separate issue to clean up old function calls across sirepo, sirepo-component, sirepo-lattice and sirepo-command.